Content/posts/5-5-released.md

```diff
@@ -0,0 +1,110 @@
+with JavaScript promises when you're using a new version of
+[JavaScriptKit](https://github.com/swiftwasm/JavaScriptKit) that was recently tagged. See the corresponding
+section below for more details.
+
+Since multi-threading in WebAssembly is still not supported across all browsers
+([Safari is the only one lagging behind](https://webassembly.org/roadmap/)), this release of
+SwiftWasm doesn't include the Dispatch library and ships with single-threaded cooperative executor. This means
+that `actor` declarations in your code will behave as plain reference type and will all be scheduled
+on the main thread. You should rely on scheduling your CPU-bound work manually with the [Web Workers
+API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers), using
+JavaScriptKit or delegating to raw JavaScript if this is needed.
+
+Additionally, 5.5.0 is the first release of SwiftWasm that supports Apple Silicon natively.
+Use the latest version of [`carton`](https://github.com/swiftwasm/carton) (0.12.0), which will download the native distribution
+automatically on supported hardware.
+
+## New JavaScriptKit runtime
+
+Accompanying 0.11 release of [JavaScriptKit](https://github.com/swiftwasm/JavaScriptKit) adds
+support for `async`/`await` and `JSPromise` integration. Now instances of this
+class have an effectful `async` property `value`. Here's example code that shows you how
+can `fetch` browser API be used without callbacks:
+
+```swift
+import JavaScriptKit
+import JavaScriptEventLoop
+
+// This line is required for `JSPromise.value` to work.
+JavaScriptEventLoop.installGlobalExecutor()
+
+private let jsFetch = JSObject.global.fetch.function!
+func fetch(_ url: String) -> JSPromise {
+    JSPromise(jsFetch(url).object!)!
+}
+
+struct Response: Decodable {
+    let uuid: String
+}
+
+let alert = JSObject.global.alert.function!
+let document = JSObject.global.document
+
+var asyncButtonElement = document.createElement("button")
+asyncButtonElement.innerText = "Fetch UUID demo"
+asyncButtonElement.onclick = .object(JSClosure { _ in
+    Task {
+        do {
+            let response = try await fetch("https://httpbin.org/uuid").value
+            let json = try await JSPromise(response.json().object!)!.value
+            let parsedResponse = try JSValueDecoder().decode(Response.self, from: json)
+            alert(parsedResponse.uuid)
+        } catch {
+            print(error)
+        }
+    }
+
+    return .undefined
+})
+
+_ = document.body.appendChild(asyncButtonElement)
+```
+
+Also, in this version of JavaScriptKit we're simplifying the `JSClosure` API. You no longer need to
+release instances of this class manually, they're automatically garbage-collected by the browser
+after your Swift code no longer has any references to them. This is achieved with the new
+`FinalizationRegistry` Web API, for which we had to significantly increase minimum browser versions
```

Comment:
Could you note that there is still a cyclic reference issue. e.g.

```
var button = document.createElement("button")
button.onclick = .object(JSClosure { [button] in
  // this capture makes a reference cycle
  print(button)
})
```

```
┌─> JSObject (button in Swift) -> HTMLElement (button in JS) ────┐
└── JSClosure (onclick in Swift) <─> Closure (onclick in JS) <───┘

```

In this case, when `button` element will be disappeared from the main DOM tree, it cannot be deallocated, and the onlick closure will be still referenced. So programmers have to cut the cycle by themselves using `weak` reference.
So we need to mention that `FinalizationRegistry` doesn't solve the all GC problem but it only solves dangling closure issue.